File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    
    clothes = [1] * (n+2)
    for l in lost :
        clothes[l] -= 1
    for r in reserve :
        clothes[r] += 1
    
    for i in range(1, n+1) :
        if clothes[i] > 0 :
            answer += 1
        else :
            if clothes[i-1] > 1 :
                clothes[i-1] -= 1
                clothes[i] += 1
                answer += 1
            elif clothes[i+1] > 1 :
                clothes[i+1] -= 1
                clothes[i] += 1
                answer += 1
    
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    
    vector<int> v(n+2, 1);
    v[0] = 0;
    v[n+1] = 0;
    
    for (int l : lost) {
        v[l] -= 1;
    }
    for (int r : reserve) {
        v[r] += 1;
    }
    
    for (int i=1; i<n+1; i++) {
        if (v[i] > 0) {
            answer += 1;
        } else {
            if (v[i-1] > 1) {
                v[i-1] -= 1;
                v[i] += 1;
                answer += 1;
            } else if (v[i+1] > 1) {
                v[i+1] -= 1;
                v[i] += 1;
                answer += 1;
            }
        }
    }
    
    return answer;
}